src/main/java/gov/cabinetoffice/gap/adminbackend/mappers/SchemeMapper.java

```diff
@@ -1,40 +1,144 @@
+    }
 
-    List<SchemeEntity> schemeDtoListToEntityList(List<SchemeDTO> schemeDto);
+    @Autowired
+    public final void setEncryptionService(AwsEncryptionServiceImpl encryptionService) {
+        this.encryptionService = encryptionService;
+    }
+
+    @Autowired
+    public final void setWebClientBuilder(WebClient.Builder webClientBuilder) {
+        this.webClientBuilder = webClientBuilder;
+    }
+
+    @Autowired
+    public final void setLambdaSecretConfigProperties(LambdaSecretConfigProperties lambdaSecretConfigProperties) {
+        this.lambdaSecretConfigProperties = lambdaSecretConfigProperties;
+    }
+
+    @Autowired
+    public final void setUserServiceConfig(UserServiceConfig userServiceConfig) {
+        this.userServiceConfig = userServiceConfig;
+    }
+
+    public SchemeDTO schemeEntityToDto(SchemeEntity schemeEntity) {
+        if ( schemeEntity == null ) {
+            return null;
+        }
+
+        SchemeDTO.SchemeDTOBuilder schemeDTO = SchemeDTO.builder();
+
+        schemeDTO.schemeId( schemeEntity.getId() );
+        schemeDTO.ggisReference( schemeEntity.getGgisIdentifier() );
+        schemeDTO.contactEmail( schemeEntity.getEmail() );
+        schemeDTO.funderId( schemeEntity.getFunderId() );
+        schemeDTO.name( schemeEntity.getName() );
+        if ( schemeEntity.getVersion() != null ) {
+            schemeDTO.version( String.valueOf( schemeEntity.getVersion() ) );
+        }
+        schemeDTO.createdDate( schemeEntity.getCreatedDate() );
+        schemeDTO.createdBy( schemeEntity.getCreatedBy() );
+
+        // Slightly strange logic here but if "last updated by" is null then we want to return "created by" instead.
+        // Considering the above, it doesn't make sense to potentially return mismatched updaters/update dates
+        // SO, if last updated by is not null then return this pair of values, otherwise return created by and created date
+        final boolean isLastUpdatedBySet = schemeEntity.getLastUpdatedBy() != null;
+        final int lastUpdatedBy  = isLastUpdatedBySet ? schemeEntity.getLastUpdatedBy() : schemeEntity.getCreatedBy();
+        final String lastUpdatedByEmail = userService.getGrantAdminById(lastUpdatedBy)
+                .map(admin -> {
+                    final String sub = admin.getGapUser().getUserSub();
+                    return getEmailAddressForSub(sub);
+                })
+                .orElse(EMPTY_EMAIL_VALUE); // Should literally never end up in here but would rather display a blank value than throw an error
+        final Instant lastUpdatedDate = isLastUpdatedBySet ? schemeEntity.getLastUpdated() : schemeEntity.getCreatedDate();
+
+        schemeDTO.lastUpdatedBy(lastUpdatedByEmail);
+        schemeDTO.lastUpdatedDate(lastUpdatedDate);
+
+        return schemeDTO.build();
+    }
+
+    private String getEmailAddressForSub(final String sub) {
+        final String url = userServiceConfig.getDomain() + "/users/emails";
```

Comment:
`/users/emails` lacks a check for the admin role attached to the user (middleware is also skipped there) . Think it's just for the lambda really. It's a minor point but we could amend:

```suggestion
        final String url = userServiceConfig.getDomain() + "/user-emails-from-subs";      
```

jwt will need to be passed in with `userServiceConfig.getCookieName()` in the request cookies eg 

`final String jwt = HelperUtils.getJwtFromCookies(request, userServiceConfig.getCookieName());`

Reply:
> `/users/emails` lacks a check for the admin role attached to the user (the auth middleware is also skipped on that endpoint) . Think it's just for the lambda / scheduler really. It's a minor security point but we could amend:
> 
> the user-service jwt will need to be passed into the request too, this is being done already in `schemeEditorService.getEmailsFromUserSubBatch` for example

Please do not call an endpoint `/user-emails-from-subs` 😭

Reply:
i'm just looking at what's in develop is all! Agree it could use a rename

Reply:
`/users/subs/emails`

Would probably be a nicer naming convention. 
I still feel like 

`/users/emails`

Is an endpoint that could (in future) be used to retrieve emails through a variety of user attributes and it's not ideal to have it limited strictly to it's lambda functionality (I am to blame as I created the endpoint so wouldn't expect you to refactor it for this)

Reply:
Just to explain why I used this endpoint - we don't have access to a request object from the mapper classes so can't go and get the JWT to send as an auth header. I would have preferred to do it the more standard way but couldn't as far as I'm aware

Reply:
We could pass the jwt in from the initial `ApplicationFormController.postApplicationForm` But yeah happy to leave it as is

Reply:
Going to look into if there's a nice way to do this without having to prop drill but it could potentially be something we fix forward